includes/background-jobs/class-sensei-scheduler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file contains Sensei_Scheduler class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Fetch the correct scheduler.
+ *
+ * @since 3.0.0
+ */
+class Sensei_Scheduler {
+	/**
+	 * Instance of the current handler.
+	 *
+	 * @var Sensei_Scheduler_Interface
+	 */
+	private static $instance;
+
+	/**
+	 * Get the instance of the Scheduler to use.
+	 *
+	 * @return Sensei_Scheduler_Interface
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			$class_name     = self::get_class();
+			self::$instance = new $class_name();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Get the class for the scheduler.
+	 *
+	 * @return string
+	 */
+	private static function get_class() {
+		if ( 0 === did_action( 'plugins_loaded' ) ) {
+			_doing_it_wrong( __METHOD__, 'Scheduler should not be used until after the plugins are loaded.', '3.0.0' );
+		}
+
+		$default_class_name = Sensei_Scheduler_WP_Cron::class;
+
+		if ( self::is_action_scheduler_available() ) {
+			$default_class_name = Sensei_Scheduler_Action_Scheduler::class;
+		}
+
+		$class_name = apply_filters( 'sensei_scheduler_class', $default_class_name );
```

Comment:
Should this also have hook documentation?

Reply:
Fixed 5a9dbba79a704d421f4d2de24eb36c5a4df5ad1b